src/Core/Services/AuthService.cs

```diff
@@ -353,27 +356,85 @@ private async Task<SymmetricCryptoKey> MakePreloginKeyAsync(string masterPasswor
                 tokenResponse.Kdf, tokenResponse.KdfIterations);
             if (_setCryptoKeys)
             {
-                await _cryptoService.SetKeyAsync(key);
-                await _cryptoService.SetKeyHashAsync(localHashedPassword);
-                await _cryptoService.SetEncKeyAsync(tokenResponse.Key);
+                if (key != null)
+                {
+                    await _cryptoService.SetKeyAsync(key);
+                }
 
-                // User doesn't have a key pair yet (old account), let's generate one for them.
-                if (tokenResponse.PrivateKey == null)
+                if (localHashedPassword != null)
                 {
-                    try
+                    await _cryptoService.SetKeyHashAsync(localHashedPassword);
+                }
+
+                if (code == null || tokenResponse.Key != null)
+                {
+                    if (tokenResponse.KeyConnectorUrl != null)
                     {
-                        var keyPair = await _cryptoService.MakeKeyPairAsync();
-                        await _apiService.PostAccountKeysAsync(new KeysRequest
+                        try
+                        {
+                            var userKeyResponse = await _apiService.GetUserKeyFromKeyConnector(tokenResponse.KeyConnectorUrl);
```

Comment:
This should use `KeyConnectorService.getAndSetKey()`, which basically just extracts this code block to a service.

Reply:
901dc76f61f185616f6f51781a06cfabe3ca403c